### DISCUSSION TITLE
Feature: Custom exception handler

### DISCUSSION THREAD

**Discussion by benedikt-bartscher** [reflex-dev/reflex#2509] [Ideas]
Currently reflex catches all exceptions during the execution of event handlers, and just uses `print()` to dump them to the console. Could we maybe add a config option to register a simple function which handles exeptions? Users could them f.e. ignore some exceptions, reraise others, or pass everything to a tracing system like sentry.

**Comment by TG199** [comment]
okay so this is what I tried:
 `def handle_exceptions(func, ignore_exceptions=(), reraise_exceptions=(), exception_handler=None):
    try:
        return func()
    except ignore_exceptions:
        pass
    except reraise_exceptions as e:
        if exception_handler:
            exception_handler(e)
        raise
    except Exception as e:
        if exception_handler:
            exception_handler(e)

        print(f"Exception occurred: {e}")
        return None`

**Reply by TG199** [reply]
@benedikt-bartscher where can I carry this out. Still trying to find my way around.

**Reply by masenf** [reply]
@TG199 i think this patch is already in progress, in https://github.com/benedikt-bartscher/reflex/pull/5

**Reply by TG199** [reply]
Oh okay, I'll try something else

**Reply by benedikt-bartscher** [reply]
@masenf is right, there is already work in progress. @TG199 we will continue to work on this mid next month and raise an upstream PR. I will try to remember reporting here when it is ready to test

**Reply by TG199** [reply]
noted

**Reply by benedikt-bartscher** [reply]
@TG199 https://github.com/reflex-dev/reflex/pull/3476 is ready for testing/review
